Source/WebKit/UIProcess/API/ios/WKWebViewIOS.mm

```diff
@@ -3404,22 +3445,28 @@ - (void)_hideFindOverlay
         if (!strongSelf)
             return;
 
-        if ([strongSelf->_findOverlay alpha])
-            return;
+        [strongSelf _removeLayerForFindOverlay];
```

Comment:
Shouldn't this be after the early return, as it was before?

Reply:
It's possible to have the overlay views outside the content view without the PageOverlay layer, but not vice-versa. This is because `_showFindOverlay` always creates the overlay views outside the content view, before creating the PageOverlay.

I think the existing structure of the code is necessary in order to guard against scenarios where the find overlay is brought up again as it's animating out – which is possible due to the async nature of the code.

So the way this code needs to be written to preserve the existing structure is:

```
bool hasOverlaysOutsideContentView = strongSelf->_findOverlaysOutsideContentView.has_value();
if (hasOverlaysOutsideContentView && [strongSelf->_findOverlaysOutsideContentView->top alpha]])
    return;

[strongSelf _removeLayerForFindOverlay];
 
if (hasOverlaysOutsideContentView) {
       [strongSelf _updateFindOverlaysOutsideContentView:^(UIView *view) {
            [view removeFromSuperview];
        }];
        strongSelf->_findOverlaysOutsideContentView.reset();
}
```